src/anm/analytics.js

```diff
@@ -0,0 +1,79 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+            } else {
+                sendViaGif(data);
+            }
+        }
+        if (!supportSendBeacon) {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (data) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = animatronUrl + '?data=' + encodeURIComponent(data);
+    };
+    if (self.enabled) {
+        if (!supportSendBeacon) {
+            engine.onDocReady(event);
```

Comment:
Omg, Anton. I'm already using sendBeacon+unload.

>Look at every sendBeacon example, they just [sum up and] push data once in a large array and send it once on window.unload

Omg, that's exactly what I do! Read the code.

If you want me to use `sendBeacon` polyfill instead of `setInterval` JUST SAY THAT.

I will use XHR polyfill. And delete `setInterval` whatsoever. Do we have something that supports CORS in IE9, or do I have to do it myself?